buildenv/docker/mkdocker.sh

```diff
@@ -0,0 +1,684 @@
+  echo "RUN cd /tmp \\"
+  echo " && $wget_O autoconf.tar.gz https://fossies.org/linux/misc/autoconf-$autoconf_version.tar.gz \\"
+  echo " && tar -xzf autoconf.tar.gz \\"
+  echo " && cd autoconf-$autoconf_version \\"
+  echo " && ./configure --build=\$(rpm --eval %{_host}) \\"
+  echo " && make \\"
+  echo " && make install \\"
+  echo " && cd .. \\"
+  echo " && rm -rf autoconf.tar.gz autoconf-$autoconf_version"
+  echo ""
+  echo "# Install gcc-7.5."
+  echo "RUN cd /usr/local \\"
+  echo " && $wget_O gcc-7.tar.xz 'https://ci.adoptopenjdk.net/userContent/gcc/gcc750+ccache.$arch.tar.xz' \\"
+  echo " && tar -xJf gcc-7.tar.xz --strip-components=1 \\"
+  echo " && ln -sf ../local/bin/gcc-7.5 /usr/bin/gcc \\"
+  echo " && ln -sf ../local/bin/g++-7.5 /usr/bin/g++ \\"
+  echo " && ln -sf gcc /usr/bin/cc \\"
+  echo " && ln -sf g++ /usr/bin/c++ \\"
+  echo " && rm -f gcc-7.tar.xz"
+  echo ""
+  local ant_version=1.10.5
+  echo "# Install ant."
+  echo "RUN cd /tmp \\"
+  echo " && $wget_O ant.zip https://archive.apache.org/dist/ant/binaries/apache-ant-$ant_version-bin.zip \\"
+  echo " && unzip -q ant.zip -d /opt \\"
+  echo " && ln -s apache-ant-$ant_version /opt/ant \\"
+  echo " && ln -s /opt/ant/bin/ant /usr/bin/ant \\"
+  echo " && $wget_O ant-contrib.tar.gz https://sourceforge.net/projects/ant-contrib/files/ant-contrib/1.0b3/ant-contrib-1.0b3-bin.tar.gz \\"
+  echo " && tar -xzf ant-contrib.tar.gz \\"
+  echo " && mv ant-contrib/ant-contrib-1.0b3.jar /opt/ant/lib \\"
+  echo " && rm -rf ant-contrib ant.zip ant-contrib.tar.gz"
+  echo ""
+  local git_version=2.5.3
+  echo "# Install git."
+  echo "RUN cd /tmp \\"
+  echo " && $wget_O git.tar.gz https://www.kernel.org/pub/software/scm/git/git-$git_version.tar.gz \\"
+  echo " && tar -xzf git.tar.gz \\"
+  echo " && cd git-$git_version \\"
+  echo " && make prefix=/usr/local all \\"
+  echo " && make prefix=/usr/local install \\"
+  echo " && cd .. \\"
+  echo " && rm -rf git.tar.gz git-$git_version"
+if [ $arch = ppc64le ] ; then
+  # make 4.2 requires automake 1.15
+  echo ""
+  local automake_version=1.15
+  echo "# Update automake."
+  echo "RUN cd /tmp \\"
+  echo " && $wget_O automake.tar.gz http://ftp.gnu.org/gnu/automake/automake-$automake_version.tar.gz \\"
+  echo " && tar -xzf automake.tar.gz \\"
+  echo " && cd automake-$automake_version \\"
+  echo " && ./configure \\"
+  echo " && make \\"
+  echo " && make install \\"
+  echo " && cd .. \\"
+  echo " && rm -rf automake.tar.gz automake-$automake_version"
+  echo ""
+  local make_version=4.2
+  echo "# Update make."
+  echo "RUN cd /tmp \\"
+  echo " && $wget_O make.tar.gz https://github.com/mirror/make/archive/$make_version.tar.gz \\"
+  echo " && tar -xzf make.tar.gz \\"
+  echo " && cd make-$make_version \\"
+  echo " && ACLOCAL_PATH=/usr/share/aclocal autoreconf -i \\"
+  echo " && ./configure \\"
+  echo " && make update \\"
+  echo " && make \\"
+  echo " && make install \\"
+  echo " && ln -s make /usr/local/bin/gmake \\"
```

Comment:
I think something is wrong here. The image runs make 3.82 and there is no `/usr/local/bin/gmake` file.

Reply:
This is only used on ppc64le where the version of make available via apt appears to hang during a build.
```
keithc@up16keithc1:~$ docker run --rm -it openj9/cent7
[root@b645b703025b /]# which make
/usr/local/bin/make
[root@b645b703025b /]# make -v
GNU Make 4.2
Built for powerpc64le-unknown-linux-gnu
Copyright (C) 1988-2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

Reply:
The tests require make 4.1 or later. The cent7/x86 image contains make 3.82

Reply:
I'll fix that, but where does it say the tests need 4.1 (besides perhaps ppc64le)?

Reply:
The test requirements are here https://github.com/eclipse/openj9/blob/master/test/docs/Prerequisites.md

Reply:
Updated to install make 4.1 on CentOS for x86_64.